src/cmd/troubleshoot/logger.go

```diff
@@ -23,26 +23,31 @@ const (
 	colorYellow = "\033[33m"
 	colorReset  = "\033[0m"
 
-	levelNewTest = 1
-	levelSuccess = 2
-	levelWarning = 3
-	levelError   = 4
+	levelNewTest     = 1
+	levelSuccess     = 2
+	levelWarning     = 3
+	levelError       = 4
+	levelNewDynakube = 5
 )
 
 type troubleshootLogger struct {
-	logger logr.Logger
+	logger  logr.Logger
+	subTest bool
 }
 
-func newTroubleshootLogger(testName string) logr.Logger {
+func newTroubleshootLogger(testName string, subTest bool) logr.Logger {
```

Comment:
I'd prefer having multiple constructors instead of passing a flag.
I.e.:
```
func newTroubleshootLogger(testName string) logr.Logger {
  return logger
}
func newSubtestTroubleshootLogger(testName string) logr.Logger {
  logger = newTroubleshootLogger(testName)
  logger.subTest = true
  return logger
}
```

Reply:
instead of using a flag to change the behaviour, using inheritence, or "embedding" in go, would be better as it upholds the open-closed principle